packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -0,0 +1,670 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  elementTypeAcceptingRef,
+  refType,
+  unstable_capitalize as capitalize,
+  unstable_useForkRef as useForkRef,
+} from '@mui/utils';
+import { useButton } from '@mui/base/ButtonUnstyled';
+import composeClasses from '@mui/base/composeClasses';
+import { useThemeProps, alpha } from '@mui/system';
+import { MD3ColorSchemeTokens, styled } from '../styles';
+import buttonClasses, { getButtonUtilityClass } from './buttonClasses';
+import { ButtonProps, ExtendButton, ButtonTypeMap, ButtonOwnerState } from './Button.types';
+
+const useUtilityClasses = (styleProps: ButtonOwnerState) => {
+  const {
+    classes,
+    color,
+    disabled,
+    disableElevation,
+    focusVisible,
+    focusVisibleClassName,
+    fullWidth,
+    size,
+    variant,
+  } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      variant,
+      `color${capitalize(color ?? '')}`,
+      `size${capitalize(size ?? '')}`,
+      disableElevation && 'disableElevation',
+      fullWidth && 'fullWidth',
+    ],
+    label: ['label'],
+    startIcon: ['startIcon', `iconSize${capitalize(size ?? '')}`],
+    endIcon: ['endIcon', `iconSize${capitalize(size ?? '')}`],
+  };
+
+  const composedClasses = composeClasses(slots, getButtonUtilityClass, classes);
+
+  if (focusVisible && focusVisibleClassName) {
+    composedClasses.root += ` ${focusVisibleClassName}`;
+  }
+
+  return composedClasses;
+};
+
+const commonIconStyles = ({ size }: ButtonOwnerState) => ({
+  color: 'var(--md-comp-button-icon-color)',
+  ...(size === 'small' && {
+    '& > *:nth-of-type(1)': {
+      fontSize: 18,
+    },
+  }),
+  ...(size === 'medium' && {
+    '& > *:nth-of-type(1)': {
+      fontSize: 20,
+    },
+  }),
+  ...(size === 'large' && {
+    '& > *:nth-of-type(1)': {
+      fontSize: 22,
+    },
+  }),
+});
+
+export const ButtonRoot = styled('button', {
+  name: 'MuiButton',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      styles[`${ownerState.variant}${capitalize(ownerState.color)}`],
+      styles[`size${capitalize(ownerState.size)}`],
+      styles[`${ownerState.variant}Size${capitalize(ownerState.size)}`],
+      ownerState.color === 'inherit' && styles.colorInherit,
+      ownerState.disableElevation && styles.disableElevation,
+      ownerState.fullWidth && styles.fullWidth,
+    ];
+  },
+})<{ ownerState: ButtonOwnerState }>(({ ownerState, theme }) => {
+  const containerColor = {
+    elevated: `linear-gradient(0deg, rgba(103, 80, 164, 0.05), rgba(103, 80, 164, 0.05)), ${
+      (theme.vars || theme).sys.color.surface
```

Comment:
Could we extract `(theme.vars || theme)` to a variable to avoid repeating this operation several times?

Reply:
Sure, will do it 👍

Reply:
Done, I've created a `tokens` const for this.